Rebrand/hhdb conditions

ivonne-hernandez opened the pull request:
What (if any) features are you implementing?
- Health insurance background/alternating background colors #FBF9FC
- Income sections should wind up alternating between white background and #FBF9FC
- Do you have income Yes/No - Selected circle is Mid Blue #41528C
- "+ ADD ANOTHER INCOME" - update font, color is Mid Blue #41528C

What (if anything) did you refactor?
- I updated the `OptionCardGroup` to match the comps and made it dynamic so that we could use it for the immediate needs question.
- I changed the outline color of the `current-household-member`.

Were there any issues that arose?
- After I merged the `Marianna/734/typography-updates` into my branch the `Income Type` dropdown stopped working. As a result I  removed the `SubHeaderWrapper` that was causing the issue and deleted the file all together since it was just an empty div with a `secondary-heading` css class.

1. @sydney-devine There is a `validateDOMNesting(...)` error that showed up on the landing page (`step-2`) indicating that the `div` below it cannot appear as a descendant of `p`:
<img width="1728" alt="Div cannot appear as a descendant of p" src="https://github.com/Gary-Community-Ventures/benefits-calculator/assets/86989161/1d640364-57ee-4a5c-9d34-bb0236d0520b">

2. Additionally, in looking at the other changes that were made and are already merged into this branding changes PR, we'll have to run the accessibility audit again to see where we're at in that regard. Some icons became images and vice versa so I'd like to see how those changes impacted accessibility.

Is there anything that you need from your teammate?
- Please review and let me know if you have any questions, or concerns.

Demo of the refactored `HouseholdDataBlock`:
https://github.com/Gary-Community-Ventures/benefits-calculator/assets/86989161/34fb123e-fbdb-4b99-abcd-f6c70e09def2

Demo of the refactored `Step-9` for Immediate Needs:
https://github.com/Gary-Community-Ventures/benefits-calculator/assets/86989161/cd4ac42d-e8b5-44c2-b510-897c3406583e

CalebPena commented on src/Components/Confirmation/Confirmation.js:
```diff
@@ -582,15 +583,17 @@ const Confirmation = () => {
     const allOtherSelectedOptions = Object.entries(hHMemberHealthInsurance).filter(
       (hHMemberInsEntry) => hHMemberInsEntry[1] === true,
     );
+    const youVsThemHealthInsuranceOptions =
+      hhMemberIndex === 1 ? healthInsuranceOptions.you : healthInsuranceOptions.them;
```
I see you fixing bugs that no one knew existed 🔥

CalebPena commented on src/Components/DropdownMenu/DropdownMenu.tsx:
```diff
@@ -21,27 +21,27 @@ interface ComponentDetails {
-  householdData: HouseholdData;
-  setHouseholdData: any;
+  memberData: HouseholdData;
+  setMemberData: any;
```
I think that we can get rid of this `any`
```suggestion
  setMemberData: (member: HouseholdData) => void;
```

CalebPena commented on src/Components/HouseholdDataBlock/HouseholdDataBlock.js:
```diff
@@ -104,24 +107,23 @@ const HouseholdDataBlock = ({ handleHouseholdDataSubmit }) => {
     if (personIndex === 1) {
       return (
-        <>
+        <Box sx={{ margin: '1rem 0 1rem 0' }}>
```
CSS trick:
```suggestion
        <Box sx={{ margin: '1rem 0' }}>
```
If you only add 2 arguments to `margin` or `padding` the first argument is the top/bottom, and the second is left/right.

CalebPena commented on src/Components/HouseholdDataBlock/HouseholdDataBlock.js:
```diff
@@ -422,31 +412,67 @@ const HouseholdDataBlock = ({ handleHouseholdDataSubmit }) => {
-        <p className="household-data-q-underline"></p>
-      </>
+      <Box className="section-container">
+        <Stack sx={{ margin: '3rem 0 3rem 0', padding: '3rem 0 3rem 0' }} className="section">
```
You can use the margin CSS trick here as well.

CalebPena commented on src/Components/HouseholdDataBlock/HouseholdDataBlock.js:
```diff
@@ -422,31 +412,67 @@ const HouseholdDataBlock = ({ handleHouseholdDataSubmit }) => {
-        <p className="household-data-q-underline"></p>
-        {householdData.hasIncome && createPersonIncomeBlock(submittedCount)}
+        <Box>
+          <Stack sx={{ margin: '3rem 0 3rem 0', padding: '3rem 0 3rem 0' }}>
```
Same CSS trick

CalebPena commented on src/Components/IncomeBlock/IncomeBlock.css:
```diff
@@ -16,3 +16,26 @@
 }
+
+.add-another-income-button.MuiButton-root {
+  color: #41528c;
```
Style controller

CalebPena commented on src/Components/IncomeBlock/IncomeBlock.css:
```diff
@@ -16,3 +16,26 @@
+}
+
+.add-another-income-button.MuiButton-root:hover {
+  background-color: #ecdeed;
```
Style controller

CalebPena commented on src/Components/IncomeBlock/IncomeQuestion.js:
```diff
@@ -109,6 +111,7 @@ const IncomeQuestion = ({
   };
 
   const handleIncomeStreamsSelectChange = (event, index) => {
+    console.log(`in here`);
```
I think that you want to remove this.

CalebPena commented on src/Components/IncomeBlock/IncomeQuestion.js:
```diff
@@ -159,6 +162,7 @@ const IncomeQuestion = ({
   };
 
   const createIncomeStreamsDropdownMenu = (incomeStreamName, index) => {
+    //error in here
```
Can you explain this comment? Is this where the error for the drop down occurs?

CalebPena commented on src/Components/OptionCardGroup/OptionCardGroup.css:
```diff
@@ -1,19 +1,26 @@
-  background-color: #e5f4f8;
-  border: 1.2px solid rgba(0, 150, 176, 0.7);
-  box-shadow: 0px 1px 5px 0px #00000040;
+  background-color: #293457;
```
Style controller

CalebPena commented on src/Components/OptionCardGroup/OptionCardGroup.css:
```diff
@@ -43,11 +50,17 @@
+  .checkmark {
+    height: 1.5rem;
+    width: 1.5rem;
+    margin: -32px 0.5rem 0.5rem 0;
```
Can we use rem here:
```suggestion
    margin: -2rem 0.5rem 0.5rem 0;
```

CalebPena commented on src/Components/OptionCardGroup/OptionCardGroup.js:
```diff
@@ -48,23 +41,43 @@ const OptionCardGroup = ({ stateVariable, options, errorController }) => {
+              boxShadow: '0px 4px 4px 0px rgba(0,0,0,0.25)',
+              height: '15rem',
+              ':hover': {
+                backgroundColor: '#ECDEED',
```
Style controller

CalebPena commented on src/Components/OptionCardGroup/OptionCardGroup.js:
```diff
@@ -1,45 +1,38 @@
-    const optionCards = Object.keys(options).map((optionKey, index) => {
-      const translatedAriaLabel = intl.formatMessage({
+  const displayOptionCards = (options, stateVariable, memberData, hhMemberIndex) => {
+    const optionCards = Object.keys(options).map((key, index) => {
+      const optionKey = key;
```
```suggestion
    const optionCards = Object.keys(options).map((optionKey, index) => {
```

CalebPena commented on src/Components/OptionCardGroup/OptionCardGroup.js:
```diff
@@ -48,23 +41,43 @@ const OptionCardGroup = ({ stateVariable, options, errorController }) => {
           <Card
-            className={state[stateVariable][optionKey] ? 'selected-option-card' : 'unselected-option-card'}
-            sx={{ width: '11.25rem', height: '11.25rem', display: 'grid', placeItems: 'center' }}
+            className={memberData[stateVariable][optionKey] ? 'selected-option-card' : 'unselected-option-card'}
+            sx={{
+              display: 'flex',
+              flexDirection: 'column',
+              border: '.05px solid #6D6E71',
+              boxShadow: '0px 4px 4px 0px rgba(0,0,0,0.25)',
+              height: '15rem',
+              ':hover': {
+                backgroundColor: '#ECDEED',
+                color: '#1d1c1e',
+              },
+            }}
```
I feel like this is a lot of CSS to inline. What do you think about adding a class to encapsulate all of this CSS, seeing that we already are using classes to mark whether or not it is selected?

CalebPena reviewed (changes requested):
Nice work, only need to add the the style controller in a couple of places, and I added some optional suggestions.

ivonne-hernandez commented on src/Components/IncomeBlock/IncomeQuestion.js:
```diff
@@ -73,11 +74,9 @@ const IncomeQuestion = ({
 
     const menuItems = menuItemKeys.map((menuItemKey, i) => {
       return (
-        <SubHeaderWrapper key={menuItemKey}>
```
This `SubHeaderWrapper` broke the onClick functionality of the `menuItems` so I removed it from here as well as the file since it was just a div with a `secondary-heading` css class.

ivonne-hernandez replied:
Yes, I did.

ivonne-hernandez commented on src/Components/IncomeBlock/IncomeQuestion.js:
```diff
@@ -367,25 +366,27 @@ const IncomeQuestion = ({
-      {incomeFrequency === 'hourly' && createHoursWorkedTextField(incomeStreamName, hoursPerWeek, index)}
-      {createIncomeAmountTextfield(incomeStreamName, incomeAmount, index)}
+    <div key={index} className="section-container income-question-padding">
+      <div className={index % 2 === 0 ? 'section' : ''}>
```
This conditional alternates the pink background color.

ivonne-hernandez commented on src/Components/OptionCardGroup/OptionCardGroup.css:
```diff
@@ -43,11 +57,17 @@
+  .checkmark {
+    height: 1.5rem;
+    width: 1.5rem;
+    margin: -2rem 0.5rem 0.5rem 0;
```
The negative rem value allows it to show up without pushing the container up 2 rems.

sydney-devine commented:
> What (if any) features are you implementing?
> 
> * Health insurance background/alternating background colors #FBF9FC
> * Income sections should wind up alternating between white background and #FBF9FC
> * Do you have income Yes/No - Selected circle is Mid Blue #41528C
> * "+ ADD ANOTHER INCOME" - update font, color is Mid Blue #41528C
> 
> What (if anything) did you refactor?
> 
> * I updated the `OptionCardGroup` to match the comps and made it dynamic so that we could use it for the immediate needs question.
> 
> Were there any issues that arose?
> 
> * After I merged the `Marianna/734/typography-updates` into my branch the `Income Type` dropdown stopped working. As a result I  removed the `SubHeaderWrapper` that was causing the issue and deleted the file all together since it was just an empty div with a `secondary-heading` css class.
> * @sydney-devine There is a `validateDOMNesting(...)` error that showed up on the landing page (`step-2`) indicating that the `div` below it cannot appear as a descendant of :
> 
> <img alt="Div cannot appear as a descendant of p" width="1728" src="https://private-user-images.githubusercontent.com/86989161/290284033-1d640364-57ee-4a5c-9d34-bb0236d0520b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0OTA4NDEsIm5iZiI6MTcwMjQ5MDU0MSwicGF0aCI6Ii84Njk4OTE2MS8yOTAyODQwMzMtMWQ2NDAzNjQtNTdlZS00YTVjLTlkMzQtYmIwMjM2ZDA1MjBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEzVDE4MDIyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjMzc0YWQ2NDg3OGYyYWI5YzcwNTFhN2NiNDc2OGQ5ZGRiZTNmYTJjM2I0NTliYjBjM2Q0YjU4OWZjMjk2NTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YIoGwztz0-3bXYYjaSyK-n7qhCZW6sgiB1J_ZcA32a4">
> Is there anything that you need from your teammate?
> 
> * Please review and let me know if you have any questions, or concerns.

Hey @mexi-cano - is the Step 2 error Ivonne found here the same one you were looking at in the other PR? 

@ivonne-hernandez can I snag a screenshot of what the page looks like now?

ivonne-hernandez commented:
> > What (if any) features are you implementing?
> > 
> > * Health insurance background/alternating background colors #FBF9FC
> > * Income sections should wind up alternating between white background and #FBF9FC
> > * Do you have income Yes/No - Selected circle is Mid Blue #41528C
> > * "+ ADD ANOTHER INCOME" - update font, color is Mid Blue #41528C
> > 
> > What (if anything) did you refactor?
> > 
> > * I updated the `OptionCardGroup` to match the comps and made it dynamic so that we could use it for the immediate needs question.
> > 
> > Were there any issues that arose?
> > 
> > * After I merged the `Marianna/734/typography-updates` into my branch the `Income Type` dropdown stopped working. As a result I  removed the `SubHeaderWrapper` that was causing the issue and deleted the file all together since it was just an empty div with a `secondary-heading` css class.
> > * @sydney-devine There is a `validateDOMNesting(...)` error that showed up on the landing page (`step-2`) indicating that the `div` below it cannot appear as a descendant of :
> > 
> > <img alt="Div cannot appear as a descendant of p" width="1728" src="https://private-user-images.githubusercontent.com/86989161/290284033-1d640364-57ee-4a5c-9d34-bb0236d0520b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0OTA4NDEsIm5iZiI6MTcwMjQ5MDU0MSwicGF0aCI6Ii84Njk4OTE2MS8yOTAyODQwMzMtMWQ2NDAzNjQtNTdlZS00YTVjLTlkMzQtYmIwMjM2ZDA1MjBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEzVDE4MDIyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjMzc0YWQ2NDg3OGYyYWI5YzcwNTFhN2NiNDc2OGQ5ZGRiZTNmYTJjM2I0NTliYjBjM2Q0YjU4OWZjMjk2NTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YIoGwztz0-3bXYYjaSyK-n7qhCZW6sgiB1J_ZcA32a4">
> > Is there anything that you need from your teammate?
> > 
> > * Please review and let me know if you have any questions, or concerns.
> 
> Hey @mexi-cano - is the Step 2 error Ivonne found here the same one you were looking at in the other PR?
> 
> @ivonne-hernandez can I snag a screenshot of what the page looks like now?

@sydney-devine Let me know if you're not able to see the screenshot of the error that I attached in the PR.

sydney-devine commented:
> > > What (if any) features are you implementing?
> > > 
> > > * Health insurance background/alternating background colors #FBF9FC
> > > * Income sections should wind up alternating between white background and #FBF9FC
> > > * Do you have income Yes/No - Selected circle is Mid Blue #41528C
> > > * "+ ADD ANOTHER INCOME" - update font, color is Mid Blue #41528C
> > > 
> > > What (if anything) did you refactor?
> > > 
> > > * I updated the `OptionCardGroup` to match the comps and made it dynamic so that we could use it for the immediate needs question.
> > > 
> > > Were there any issues that arose?
> > > 
> > > * After I merged the `Marianna/734/typography-updates` into my branch the `Income Type` dropdown stopped working. As a result I  removed the `SubHeaderWrapper` that was causing the issue and deleted the file all together since it was just an empty div with a `secondary-heading` css class.
> > > * @sydney-devine There is a `validateDOMNesting(...)` error that showed up on the landing page (`step-2`) indicating that the `div` below it cannot appear as a descendant of :
> > > 
> > > <img alt="Div cannot appear as a descendant of p" width="1728" src="https://private-user-images.githubusercontent.com/86989161/290284033-1d640364-57ee-4a5c-9d34-bb0236d0520b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0OTA4NDEsIm5iZiI6MTcwMjQ5MDU0MSwicGF0aCI6Ii84Njk4OTE2MS8yOTAyODQwMzMtMWQ2NDAzNjQtNTdlZS00YTVjLTlkMzQtYmIwMjM2ZDA1MjBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEzVDE4MDIyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjMzc0YWQ2NDg3OGYyYWI5YzcwNTFhN2NiNDc2OGQ5ZGRiZTNmYTJjM2I0NTliYjBjM2Q0YjU4OWZjMjk2NTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YIoGwztz0-3bXYYjaSyK-n7qhCZW6sgiB1J_ZcA32a4">
> > > Is there anything that you need from your teammate?
> > > 
> > > * Please review and let me know if you have any questions, or concerns.
> > 
> > 
> > Hey @mexi-cano - is the Step 2 error Ivonne found here the same one you were looking at in the other PR?
> > @ivonne-hernandez can I snag a screenshot of what the page looks like now?
> 
> @sydney-devine Let me know if you're not able to see the screenshot of the error that I attached in the PR.

Oh sorry my note wasn't clear. Screenshot of what the household conditions etc. addressed in this PR looks like now. Pleeease :)

CalebPena commented on src/Assets/styleController.ts:
```diff
@@ -43,6 +50,13 @@ const themes: IThemes = {
+      '--option-card-hover-background-color': '#ECDEED',
+      '--option-card-hover-font-color': '#1D1C1E',
+      '--option-card-icon-fill-color': '#D6743F',
+      '--selected-option-card-background-color': '#293457',
```
This is the same as primary color. could you use primary color instead?

CalebPena commented on src/Assets/styleController.ts:
```diff
@@ -43,6 +50,13 @@ const themes: IThemes = {
       'font-size': '16px',
       '--main-header-color': '#B85A27',
       '--midBlue-color': '#41528C',
+      '--option-card-background-color': '#FFFFFF',
```
I don't think that we need to add anything that is just the color white to the style controller.

CalebPena commented on src/Assets/styleController.ts:
```diff
@@ -43,6 +50,13 @@ const themes: IThemes = {
       '--main-header-color': '#B85A27',
       '--midBlue-color': '#41528C',
+      '--option-card-background-color': '#FFFFFF',
+      '--option-card-hover-background-color': '#ECDEED',
```
Can you pull from the feature branch. On the feature branch, there is a variable called `hoverColor`. What do you think about calling this CSS variable `--hover-color` for consistency?

CalebPena reviewed (approved):
LGTM
